civis_jupyter_notebooks/assets/extensions/terminal.js

```diff
@@ -0,0 +1,22 @@
+define(['base/js/namespace'], function(Jupyter) {
+  function _on_load() {
+    var action = {
+      icon: 'fa-terminal',
+      help: 'Open Terminal',
+      handler: function(env) {
+        env.notebook.save_checkpoint();
+
+        // stop pop up from displaying
+        window.onbeforeunload = function(){}
```

Comment:
this is needed even though we've just saved the notebook? Seems weird.

Reply:
Yea, they wrote their own `onbeforeunload` that sets this alert ...